### DISCUSSION TITLE
تولید گوهینامه SSL به وسیله certbot

### DISCUSSION THREAD

**Discussion by alireza-s13** [Gozargah/Marzban#188] [Ideas]
سلام خدمت دوستان در صورتی که برای گرفتن گواهینامه SSL به وسیله `acme.sh` مشکلی دارید می‌توانید از طریق certbot اقدام کنید که در ادامه توضیح خواهم داد. ابتدا طبق آموزش پیش برید و در قسمت **فعال سازی TLS** از روش زیر استفاده کنید.
1- ابتدا snapd را نصب کنید. 
```bash
sudo apt install snapd
```
2- در صورتی که از نسخه قدیمی certbot استفاده می‌کنید آن را پاک کنید.
```bash
sudo apt remove certbot
```
3- سپس از طریق دستور زیر certbot را نصب کنید.
```bash
sudo snap install --classic certbot
```
4- مطمئن شوید که یکی از پورت 443 یا 80 روی آی پی سرور شما باز باشد. می‌توانید از طریق [این لینک](https://dnschecker.org/port-scanner.php) بررسی کنید.
5- دامنه خود را جایگزین your_domain و دستور زیر را اجرا کنید.
```bash
sudo certbot certonly --standalone -d your_domain
```
در صورت موفقیت آمیز بودن چنین خروجی ای خواهید داشت :
```bash
Successfully received certificate.
Certificate is saved at: /etc/letsencrypt/live/your_domain/fullchain.pem
Key is saved at:         /etc/letsencrypt/live/your_domain/privkey.pem
```
6- حالا کافی هست یک پوشه certs ایجاد و دو فایل بالا رو به پوشه certs مرزبان منتقل کنیم که کافی است دستورات زیر را اجرا کنیم:
```bash
 mkdir /var/lib/marzban/certs
cp /etc/letsencrypt/live/your_domain/fullchain.pem /var/lib/marzban/certs/fullchain.pem
cp /etc/letsencrypt/live/your_domain/privkey.pem /var/lib/marzban/certs/key.pem
```

**Comment by babirich** [comment]
توی دستور شماره 6 انتهای خط دوم بعد از certs/ چیزی نباید باشه مثل خط سوم؟

**Reply by alireza-s13** [reply]
درود خدمت شما، بله درسته
✅ اصلاح شد

**Comment by mojmoj0938** [comment]
درود و عرض ادب
چرا با وجود اینکه همه کارا رو دقیق انجام دادم ولی بازم اس اس ال نمیگیره واسم...همه چی اوکی هست ولی وقتی میخوام برم روی پنل ارور اس اس ال میده

**Reply by alireza-s13** [reply]
سلام
اگر پیام Successfully received certificate نمایش داده شد و در آدرس پوشه های گفته شده فایل ها موجود باشه نباید مشکلی وجود داشته باشه. ممکنه به خاطر کش باشه که صبر کنید یا کش مرورگر را پاک کنید حل میشه. حتی میتونید از سایت های ssl check برای بررسی استفاده کنید

**Reply by mojmoj0938** [reply]
درود...همه چی اوکی هست...بارها و بارها سرتیفیکیت گرفتم مشکلی نبوده
عجیبه که الان همه چی اوکی میشه ولی پنل مرزبان بدون اس اس ال میاد بالا بازم

**Reply by mojmoj0938** [reply]
ssl check رو چک کردم میگه مشکل اس اس ال داری ولی سرتیفیکیت رو اوکی میکنه واسه من که حتی فایل هاشم ساخته میشه
"No SSL certificates were found on elvinace.mioolla.online. Make sure that the name resolves to the correct server and that the SSL port (default is 443) is open on your server's firewall."

**Reply by mojmoj0938** [reply]
کش مرورگر هم پاک کردم...پورت ها هم آزادن

**Reply by mojmoj0938** [reply]
پورت ۴۴۳ time out میده...این یعنی پورت ۴۴۳ مشکل داره؟؟

**Reply by alireza-s13** [reply]
خب مشکل همینه. پورت 443 رو باز کنید بر اساس سیستم عامل دستورات متفاوت هست. اون موقع مشکل حل میشه

**Reply by mojmoj0938** [reply]
من هر کاری کردم نمیدونم مشکل از کجا هست که پورت ۴۴۳ باز نمیشه...
اوبنتو ۲۲ 
ufw رو فعال کردم و دسترسی ۴۴۳ و ۸۰ رو دادم بهش ولی بازم کار نمیکنه...ده بار سرور رو ریبیلد کردم دوباره نصب کردم بازم نشد
لطفا اگر شما راهی به ذهنتون میرسه در این مورد راهنمایی کنید

**Reply by alireza-s13** [reply]
خب اگر با ریبیلد فعال نمیشه مشکل از جای دیگه هست. اول اینکه چک کنید ببنید اون شرکتی که ازش سرور دارید خودشون فایروالی ندارند که روی اکانت شما فعال باشه. مثلا هتزنر چنین قابلیتی داره. 
اگر بازم حل نشد بهتره با پشتیبانی سرور مطرح کنید.

**Reply by mojmoj0938** [reply]
دقیقا من هم از هتزنر گرفتم و یه بار هم آی پیم رو مسدود کردن بخاطر netscan

**Reply by DoctorGuidance** [reply]
دادا مرزبان خودش وب سرور هستش و پورت رو بسته مهم نیست با ufw بازش کنی یا نه. توی .evn غیره 8000 رو بکنی 443 تا باز شه برا پنل، باقی پورت ها هم هرکدوم توی inbound وارد میکنی اتوماتیک باز میشن. مثلا 2053، 8443

**Comment by MMRP77** [comment]
با اجازه،اصلاح میکنم
بعد از اینکه موارد بالا رو انجام دادید
وارد فایل .env بشید
پورت443
و قسمتUVICORN_SSL_CERTFILE و UVICORN_SSL_KEYFILE باید بدین صورت ویرایش بشه:(میتونید همینو کپی کنید جایگذاری کنید)

UVICORN_SSL_CERTFILE = "/var/lib/marzban/certs/fullchain.pem"
UVICORN_SSL_KEYFILE = "/var/lib/marzban/certs/key.pem"

سیو کنید
مرزبان رو ریست کنید
و آدرس رو اینجوری سرچ کنید
https://sub.domin.com/dashboard/
قسمت sub.domin.com دامنه و یا سابدامنه خودتون رو بزنید

**Reply by xRay-v2Ray** [reply]
نشد

**Comment by xRay-v2Ray** [comment]
آقا پدرم درومد
هرکاری میکنم میگه پورت 80 در حال استفاده است
Could not bind TCP port 80 because it is already in use by another process on
this system (such as a web server). Please stop the program in question and then
try again.

چیکار کنم من ؟

**Reply by amirnikjoo** [reply]
با دستور زیر ببین چه پراسسی روی پورت 80 هست. 
lsof -i tcp:80

**Comment by Meytiz** [comment]
سلام این مدل Ssl گرفتن مشکل داره ؟ من ssl گرفتم باهاش ولی ن باز میکنه ن اجازه انتقال فایل رو میده مشکلش چیه؟

**Comment by xGavazn** [comment]
میتونید سرتیفیکیت Wildcard بگیرید // یک سرتیفیکیت برای هر ساب دامینی
https://github.com/Gozargah/Marzban/discussions/869#discussion-6367626 از اینجا**

**Comment by shahinmagic1** [comment]
در حالی که یک ساله با این گواهی کار میکنم امروز خراب شده تاریخ مصرفش تموم شده اپدیت کردم این ارور رو داد
cannot create directory ‘/var/lib/marzban/certs’: File exists
پورت 443 هستش فایوال هم فعال نیست.
گواهی رو پاک کردم از اول نصب کردم جواب نداد. گهواهی رو اپدیت کردم جواب نداد. در هر صورت کلا جواب نداد. خلاصه نمیدونم چه بلایی سرش اوردین کلا خراب شده

**Comment by Immojtaba76** [comment]
این ارور رو چجوری رفع کنم

[
![Uploading Screenshot_20240605-223920_Termius.jpg…]()
](url)

**Reply by crowlink** [reply]
درود و خسته نباشید 
به سرور مرکزی چند دامین رو بخواهیم اس اس ال دار کنیم
با چه فرمتی جلو بریم؟
